### ISSUE TITLE
Fix S1104 FP: required property type flagged as field

### ISSUE THREAD

**Issue by joepb** [SonarSource/sonar-dotnet#7230]
### Description

When using the new C# [required modifier](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required) on a property, S1104 is triggered incorrectly. 

This happens only when the type of the required property is a custom definition, not a built-in type.

### Repro steps

```cs
  public sealed record Foo
  {
      public required Bar Baz { get; init; }  // "Bar" is flagged with S1104: Fields should not have public accessibility
  }
  
  public sealed record Bar
  {
  }
```

### Expected behavior

The code is valid and should not be marked.

### Actual behavior

The type name is marked as if it is a public field, which it isn't.

### Known workarounds

None.

### Related information

* Jetbrains Rider 2022.3.3
* SonarLint plugin 8.2.0.68615
* dotnet 7.0 project
* Microsoft Windows 10.0.22621.1702

**Comment by mary-georgiou-sonarsource** [comment]
Hello @joepb!

Thanks for reporting this.
I could not reproduce this issue outside of Rider so probably it's related to SonarLint for Rider.

I will be back to you once I have more info.

Thanks!

**Comment by henryju** [comment]
Hi, we were able to investigate, and apparently, the version of Omnisharp we are using (1.39.0) is too old. I made a quick test and the issue is not raised with the latest version of Omnisharp (1.39.6). I have created a ticket to update it:
https://sonarsource.atlassian.net/browse/SLI-957

Thanks for reporting it!
